tests/quick_tests/taskflow.cc

```diff
@@ -0,0 +1,83 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2020 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+// test taskflow
+
+#include <deal.II/base/thread_management.h>
+
+DEAL_II_DISABLE_EXTRA_DIAGNOSTICS
```

Comment:
Are there many warnings to subpress?

Reply:
We do the same in `source/base/multithread_info.cc` and also for a lot of other dependencies so I am fine with this. It seems that they are pretty open to suggestions for improvements, though. It might be worth a shot disabling it, fixing the warnings we see and create a pull request.

Reply:
> Are there many warnings to subpress?

2.4.0 (which we bundle) triggers this https://github.com/taskflow/taskflow/pull/170 for me otherwise.